src/workstation-gui/app/helpers/application_helper.rb

```diff
@@ -1,2 +1,60 @@
 module ApplicationHelper
+  DEFAULT_LIMIT = 10
+  DEFAULT_PAGE = 1
+  def chef_dir
+    File.expand_path("~/.chef") # todo move this file in .chef string in module with constants
+  end
+
+  def chef_repo_storage_file
+    File.expand_path(File.join(chef_dir, "repository.json"))
+  end
+
+  def create_chef_repo_storage_file
+    fobj = File.open(chef_repo_storage_file, "w") do |f|
+      f.write({ "repositories" => [] }.to_json)
+    end
+    fobj.close
+    chef_repo_storage_file
+  end
+
+  def read_repo_file
+    File.read(chef_repo_storage_file)
+  end
```

Comment:
I see you include this module in `ApplicationController` and there are a lot of methods is written here. 

So this can become a performance issue because every controller inherits from this controller. As this method will always be in memory irrespective controller is invoking that method or not. Usually, we write only those which are needed for every request like Authentication code, pagination

Reply:
Removed this and moved it into another module helper( WorkstationHelper)